game/missiongenerator/luagenerator.py

```diff
@@ -210,6 +211,49 @@ def generate_plugin_data(self) -> None:
                     iads_element.add_key_value(property, value)
             for role, connections in node.connections.items():
                 iads_element.add_data_array(role, connections)
+        
+        # Add artillery and support units info
+        artillery_object = lua_data.add_item("artilleryGroups")
+        ground_artillery_group_collection = artillery_object.get_or_create_item("groundArtillery")
+        ship_artillery_group_collection = artillery_object.get_or_create_item("shipArtillery")
+
+        # First add all artillery units that are theater objects (mostly ships)
+        for cp in self.game.theater.control_points_for(player=True):
+            for ground_object in cp.ground_objects:
+                for group in ground_object.groups:
+                    # Check if first unit in group is ground-based or ship artillery
+                    unit = group.units[0]
+                    if unit.unit_type is None:
+                        continue
+                    if unit.unit_type.unit_class == UnitClass.ARTILLERY:
+                        ground_artillery_group = ground_artillery_group_collection.add_item()
+                        ground_artillery_group.add_key_value("groupName", group.group_name)
+                    elif unit.unit_type.unit_class in (UnitClass.CRUISER, UnitClass.DESTROYER, UnitClass.FRIGATE):
+                        # TODO: we assume that these ship classes have guns... Which might not be the case.
+                        ship_artillery_group = ship_artillery_group_collection.add_item()
+                        ship_artillery_group.add_key_value("groupName", group.group_name)
+        
+        # Add artillery that are frontline groups
+        for frontline_group in self.mission_data.player_frontline_groups:
+            if frontline_group.unit_type.unit_class is None:
+                continue
+            if frontline_group.unit_type.unit_class == UnitClass.ARTILLERY:
+                ground_artillery_group = ground_artillery_group_collection.add_item()
+                ground_artillery_group.add_key_value("groupName", frontline_group.group_name)
+            
+
+        # Add forward observer (FO) (TODO: maybe adding new flight type "Foward Observer"?)
```

Comment:
I like the idea of having a "dedicated" role to that, plus it can help keep it manageable in large MP scenarios, because I haven't tested the limits of MBot's script. It could be a user toggleable option in the plugins options, but that would necessitate a bit more LUA code to do that.

I can either take more time to update this PR with that or I can prepare another PR with that update, whatever you prefer!

Reply:
In that case I'd say let's go ahead and merge this already and introduce a separate flight-type later, because I can imagine there'll be a few quirks to this one since it's probably best to keep AI from using that flight-type, since they won't be of much use 😅